econml/tests/test_automated_ml.py

```diff
@@ -0,0 +1,235 @@
+                                     debug_log='automl_errors.log',
+                                     enable_onnx_compatible_models=True, model_explainability=True,
+                                     **AUTOML_SETTINGS_CLF)
+
+AUTOML_CONFIG_LINEAR_REG = EconAutoMLConfig(task='regression',
+                                            debug_log='automl_errors.log',
+                                            enable_onnx_compatible_models=True, model_explainability=True,
+                                            **AUTOML_SETTINGS_REG)
+
+AUTOML_CONFIG_SAMPLE_WEIGHT_REG = EconAutoMLConfig(task='regression',
+                                                   debug_log='automl_errors.log',
+                                                   enable_onnx_compatible_models=True, model_explainability=True,
+                                                   **AUTOML_SETTINGS_REG)
+
+
+def automl_model_reg():
+    return copy.deepcopy(AUTOML_CONFIG_REG)
+
+
+def automl_model_clf():
+    return copy.deepcopy(AUTOML_CONFIG_CLF)
+
+
+# Linear models are required for parametric dml
+
+def automl_model_linear_reg():
+    return copy.deepcopy(AUTOML_CONFIG_LINEAR_REG)
+
+
+# sample weighting models are required for nonparametric dml
+
+def automl_model_sample_weight_reg():
+    return copy.deepcopy(AUTOML_CONFIG_SAMPLE_WEIGHT_REG)
+
+
+# Test values
+Y, T, X, _ = ihdp_surface_B()
+
+
+@pytest.mark.automl
+class TestDML(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        subscription_id = os.getenv("SUBSCRIPTION_ID")
+        resource_group = os.getenv("RESOURCE_GROUP")
+        workspace_name = os.getenv("WORKSPACE_NAME")
+
+        auth = AzureCliAuthentication()
+
+        setAutomatedMLWorkspace(auth=auth,
+                                subscription_id=subscription_id,
+                                resource_group=resource_group, workspace_name=workspace_name)
+
+    def test_nonparam(self):
+        Y, T, X, _ = ihdp_surface_B()
+        est = \
+            AutomatedNonParamDMLCateEstimator(model_y=automl_model_reg(),
+                                              model_t=automl_model_clf(),
+                                              model_final=automl_model_reg(), featurizer=None,
+                                              discrete_treatment=True)
+        est.fit(Y, T, X)
+        _ = est.effect(X)
+
+    def test_param(self):
+        Y, T, X, _ = ihdp_surface_B()
+        est = \
+            AutomatedDMLCateEstimator(model_y=automl_model_reg(),
+                                      model_t=GradientBoostingClassifier(),
+                                      model_final=RandomForestRegressor(),
+                                      featurizer=None,
+                                      discrete_treatment=True)
+        est.fit(Y, T, X)
+        _ = est.effect(X)
+
+    def test_forest_dml(self):
+        """Testing accuracy of forest DML is reasonable"""
```

Comment:
This doesn't test accuracy in any way; change the comment to explain what the test actually does.